test_framework/terraform/aws/centos/main.tf

```diff
@@ -268,4 +284,95 @@ resource "aws_ebs_volume" "lh_aws_hdd_volume" {
   tags = {
     Name = "lh-aws-hdd-volume-${random_string.random_suffix.id}-${count.index}"
   }
+}
+
+# Create load balancer for rancher
+resource "aws_lb_target_group" "lh_aws_lb_tg_443" {
+
+  count = var.create_load_balancer ? 1 : 0
+
+  name     = "lh-aws-lb-tg-443-${random_string.random_suffix.id}"
+  port     = 443
+  protocol = "TCP"
+  vpc_id   = aws_vpc.lh_aws_vpc.id
+  health_check {
+    protocol = "TCP"
+    port = "80"
+    healthy_threshold = 3
+    unhealthy_threshold = 3
+    interval = 10
+  }
+}
+
+resource "aws_lb_target_group" "lh_aws_lb_tg_80" {
+
+  count = var.create_load_balancer ? 1 : 0
+
+  name     = "lh-aws-lb-tg-80-${random_string.random_suffix.id}"
+  port     = 80
+  protocol = "TCP"
+  vpc_id   = aws_vpc.lh_aws_vpc.id
+  health_check {
+    protocol = "TCP"
+    port = "traffic-port"
+    healthy_threshold = 3
+    unhealthy_threshold = 3
+    interval = 10
+  }
+}
+
+resource "aws_lb" "lh_aws_lb" {
+
+  count = var.create_load_balancer ? 1 : 0
+
+  depends_on = [
+    aws_internet_gateway.lh_aws_igw
+  ]
+
+  name               = "lh-aws-lb-${random_string.random_suffix.id}"
+  internal           = false
+  load_balancer_type = "network"
+  subnets            = [
+    aws_subnet.lh_aws_public_subnet.id
+  ]
+}
+
+resource "aws_lb_listener" "lh_aws_lb_listener_443" {
+
+  count = var.create_load_balancer ? 1 : 0
+
+  depends_on = [
+    aws_lb.lh_aws_lb,
+    aws_lb_target_group.lh_aws_lb_tg_443
+  ]
+
+  load_balancer_arn = aws_lb.lh_aws_lb[0].arn
+  port              = "443"
+  protocol          = "TCP"
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.lh_aws_lb_tg_443[0].arn
+  }
+
+}
+
+resource "aws_lb_listener" "lh_aws_lb_listener_80" {
+
+  count = var.create_load_balancer ? 1 : 0
+
+  depends_on = [
+    aws_lb.lh_aws_lb,
+    aws_lb_target_group.lh_aws_lb_tg_80
+  ]
+
+  load_balancer_arn = aws_lb.lh_aws_lb[0].arn
+  port              = "80"
```

Comment:
Do we need 80 port LB? In main.py, it just used https to communicate with the Rancher server.

Reply:
Modified.